pkg/scheduler/objects/application.go

```diff
@@ -1920,6 +1947,26 @@ func (sa *Application) cleanupAsks() {
 	sa.sortedRequests = nil
 }
 
+func (sa *Application) GetAppSummary(rmID string) string {
+	var res string
+	sa.RLock()
+	defer sa.RUnlock()
+
+	res = "{"
+
+	res += "\"appId\":\"" + sa.ApplicationID + "\","
+	res += "\"submissionTime\":\"" + strconv.FormatInt(sa.SubmissionTime.UnixMilli(), 10) + "\","
+	res += "\"startTime\":\"" + strconv.FormatInt(sa.StartTime().UnixMilli(), 10) + "\","
+	res += "\"finishTime\":\"" + strconv.FormatInt(sa.FinishedTime().UnixMilli(), 10) + "\","
+	res += "\"user\":\"" + sa.GetUser().User + "\","
+	res += "\"queue\":\"" + sa.GetQueuePath() + "\","
+	res += "\"state\":\"" + sa.stateMachine.Current() + "\","
+	res += "\"rmID\":\"" + rmID + "\","
+	res += "\"resourceUsage\":" + sa.usedResourceTracker.GetResourceUsageSummary()
+	res += "}"
+	return res
```

Comment:
It's weird to see all these formatting here. I do believe the `Application` object should not be concerned about formatting,  that should be delegated to the caller.

One approach is to just call the Get* methods to retrieve the relevant piece of data. Missing getters can be added.

If it's necessary to return a snapshot of these data guarded by locks, you can create an `AppSummary` type which contains all these information and return that.

Reply:
Thanks, will be addressed in next rev.